### PR TITLE
samples: ncp: Added Spinel logging backend as default.

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
Spinel logging backend was set as a default one.
Also new overlay changing log levels of all modules to INFO
was introduced, as by default all levels are set to CRIT.

Signed-off-by: Kamil Kasperczyk <kamil.kasperczyk@nordicsemi.no>

### CODE REVIEW

**Comment by greg-fer** [comment]
@kkasperczyk-no , is this task related to KRKNWK-6767 ?

**Comment by kkasperczyk-no** [comment]
> @kkasperczyk-no , is this task related to KRKNWK-6767 ?

No, to the KRKNWK-7076

File: samples/openthread/ncp/prj.conf [hunk 1]
```diff
@@ -69,9 +71,6 @@ CONFIG_NET_PKT_TX_COUNT=16
 CONFIG_NET_BUF_RX_COUNT=80
 CONFIG_NET_BUF_TX_COUNT=80
 
-# Use UART_1 for NCP
-CONFIG_OPENTHREAD_NCP_SPINEL_ON_UART_DEV_NAME="UART_0"
```
Comment (edmont) [inline samples/openthread/ncp/prj.conf:73]: This is not needed anymore?

Comment (kkasperczyk-no) [reply samples/openthread/ncp/prj.conf:73]: Yes, it is set to UART_0 by default

File: samples/openthread/ncp/overlay-logging.conf [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#
+# Copyright (c) 2020 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+#
+# option for configuring log level in sample module
```
Comment (edmont) [inline samples/openthread/ncp/overlay-logging.conf:6]: Please consider starting the comment lines with upper case. Also remove the # for the empty line above.

File: samples/openthread/ncp/prj.conf [hunk 3]
```diff
@@ -18,8 +18,10 @@ CONFIG_NET_LOG=y
 CONFIG_LOG=y
 CONFIG_NET_STATISTICS=y
 CONFIG_PRINTK=y
-CONFIG_OT_NCP_LOG_LEVEL_DBG=y
```
Comment (edmont) [inline samples/openthread/ncp/prj.conf:21]: Maybe leave this line here with the default value? For consistency with `overlay-logging.conf`.

**Review by MarekPorwisz — APPROVED** [review]

**Comment by kkasperczyk-no** [comment]
@rlubos it finally passed CI, so I think it's ready to be merged.

**Review by rlubos — APPROVED** [review]

**Pull request merged**